common/src/main/java/org/opensearch/ml/common/transport/connector/MLCreateConnectorInput.java

```diff
@@ -93,6 +93,15 @@ public MLCreateConnectorInput(
             if (protocol == null) {
                 throw new IllegalArgumentException("Connector protocol is null");
             }
+            if (description == null) {
```

Comment:
I think description should not be a required field. May be we need to update the documentation from where you got this reference.

Reply:
Yes, I also had the same opinion while coding it. Since it is clearly marked is as required field in documentation, thought of discussing it on the review if someone raise the same concern. Similarly, I feel fields "parameters" and "credential" can also be null, if there is any parameters then user can configure otherwise can keep these fields as not configured. Also, let's wait for input from @rbhavna on "requestBody", that was intentionally kept as null and even explicit assertions for null were there in unit test cases. Same time documentation says, it's a required field.

I didn't add validation for backend_roles, access_mode, add_all_backend_roles though these were marked as required fields in documentation. These are inter related fields and corresponding validations exist in TransportCreateConnectorAction.validateRequest4AccessControl() and TransportCreateConnectorAction.validateSecurityDisabledOrConnectorAccessControlDisabled(). If none of these parameters configured we don't throw exception, but if one configure then others also has to configure properly(accordingly).

I referred the documentation - https://opensearch.org/docs/latest/ml-commons-plugin/remote-models/blueprints/#configuration-parameters

I wanted to discuss this before raising PR, but I thought it will be more easy to understand when reviewers see the code changes.

Reply:
Yeah I agree parameters could be empty too. But not credentials. For connector, credentials are required. May be we can apply validation for Credentials but not for parameters?

Reply:
Yeah, I think we add default role if role is not provided. So no need to have validation for roles as well. @rbhavna please correct me if I'm wrong.

Reply:
@akolarkunnu Thanks for your contribution. And yes that is correct. We don't need any validation here for roles. We will validate them in transport layer after deciding if its security enabled or disabled cluster and accordingly validate the fields. So we can skip it here.

Reply:
As @dhrubo-os mentioned, `description` and `parameters` can be optional too. Looks like opensearch documentation needs to be updated here https://opensearch.org/docs/latest/ml-commons-plugin/remote-models/blueprints/#configuration-parameters

Reply:
ok, so new validation is added only for "credentials". All others removed.